kolibri/core/content/tasks.py

```diff
@@ -35,3 +45,493 @@ def importcontentfromdisk(**kwargs):
+            check_for_cancel=check_for_cancel,
+        )
+
+        # Make some real-time updates to the metadata
+        job = get_current_job()
+
+        # Signal to UI that the DB-downloading step is done so it knows to display
+        # progress correctly
+        job.update_progress(0, 1.0)
+        job.extra_metadata["database_ready"] = True
+
+        # Add the channel name if it wasn't added initially
+        if job and job.extra_metadata.get("channel_name", "") == "":
+            job.extra_metadata["channel_name"] = get_channel_name(channel_id)
+
+        job.save_meta()
+
+        # Skip importcontent step if updating and no nodes have changed
+        if is_updating and (node_ids is not None) and len(node_ids) == 0:
+            pass
+        else:
+            call_command(
+                "importcontent",
+                "disk",
+                channel_id,
+                directory,
+                drive_id=drive_id,
+                node_ids=node_ids,
+                exclude_node_ids=exclude_node_ids,
+                update_progress=update_progress,
+                check_for_cancel=check_for_cancel,
+            )
+
+
+def validate_startremotechannelimport(request, request_data):
+    task = validate_remote_import_task(request, request_data)
+    task.update({"type": "REMOTECHANNELIMPORT"})
+    task["extra_metadata"] = copy.deepcopy(task)
+    return task
+
+
+@register_task(
+    validator=validate_startremotechannelimport,
+    cancellable=True,
+    permission_classes=[CanManageContent],
+)
+def startremotechannelimport(channel_id=None, baseurl=None, peer_id=None):
+    call_command(
+        "importchannel",
+        "network",
+        channel_id,
+        baseurl=baseurl,
+        peer_id=peer_id,
+    )
+
+
+def validate_startremotecontentimport(request, request_data):
+    task = validate_remote_import_task(request, request_data)
+    task.update({"type": "REMOTECONTENTIMPORT"})
+    task["extra_metadata"] = copy.deepcopy(task)
+    return task
+
+
+@register_task(
+    validator=validate_startremotecontentimport,
+    track_progress=True,
+    cancellable=True,
+    permission_classes=[CanManageContent],
+)
+def startremotecontentimport(
+    channel_id=None,
+    baseurl=None,
+    peer_id=None,
+    node_ids=None,
+    exclude_node_ids=None,
+):
+    call_command(
+        "importcontent",
+        "network",
+        channel_id,
+        baseurl=baseurl,
+        peer_id=peer_id,
+        node_ids=node_ids,
+        exclude_node_ids=exclude_node_ids,
+    )
+
+
+def validate_startdiskexport(request, request_data):
+    task = validate_local_export_task(request, request_data)
+    task.update({"type": "DISKCONTENTEXPORT"})
+    task["extra_metadata"] = copy.deepcopy(task)
+    return task
+
+
+@register_task(
+    validator=validate_startdiskexport,
+    track_progress=True,
+    cancellable=True,
+    permission_classes=[CanManageContent],
+)
+def startdiskexport(
```

Comment:
`startdiskbulkexport` will be replaced with the bulk version of `startdiskexport`. Here bulk version means frontend will be sending a list of `startduskexport` tasks to our new tasks create api.